### PR TITLE
Depv7

### PR DESCRIPTION
**Pull request by pkofod**

### CODE REVIEW

**Comment by pkofod** [comment]
@chriselrod thanks for the effort! I'm taking over on this branch, but your commit history will still live on, so you should still get credit in the git history!

File: src/complex_real.jl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"Convert a real array of size 2 x dims to a complex array of size dims"
+function real_to_complex(A::AbstractArray{T}) where T<:Real
```
Comment (anriseth) [inline src/complex_real.jl]: `real_to_complex` does not want to die :laughing:

File: src/interface.jl [hunk 2]
```diff
@@ -152,8 +154,9 @@ value!!(obj::NonDifferentiable{TF, TX}, x) where {TF<:AbstractArray, TX} = value
 value!!(obj::OnceDifferentiable{TF, TDF, TX}, x) where {TF<:AbstractArray, TDF, TX} = value!!(obj, obj.F, x)
 function value!!(obj, F, x)
     obj.f(F, x)
-    copy!(obj.x_f, x)
+    copyto!(obj.x_f, x)
     obj.f_calls .+= 1
+    obj.f_calls
```
Comment (anriseth) [inline src/interface.jl]: Is there a reason why we return this value?

File: src/interface.jl [hunk 3]
```diff
@@ -164,25 +167,29 @@ function _clear_f!(d::NLSolversBase.AbstractObjective)
         d.F = typeof(d.F)(NaN)
     end
     d.x_f .= eltype(d.x_f)(NaN)
+    d.x_f
```
Comment (anriseth) [inline src/interface.jl]: Is there a change in 0.7 that returns the right hand side of the dot-assignment if it's the last line?

File: src/objective_types/constraints.jl [hunk 4]
```diff
@@ -233,7 +233,7 @@ struct UnquotedString
 end
 Base.show(io::IO, uqstr::UnquotedString) = print(io, uqstr.str)
 
-Base.array_eltype_show_how(a::Vector{UnquotedString}) = false, ""
+#Base.array_eltype_show_how(a::Vector{UnquotedString}) = false, ""
```
Comment (anriseth) [inline src/objective_types/constraints.jl]: Comment and say it's a weird remnant from `IPNewton`? (Or just delete it)

File: test/constraints.jl [hunk 5]
```diff
@@ -14,8 +14,7 @@
         io = IOBuffer()
         show(io, cb)
         s = String(take!(io))
-        @test s == "ConstraintBounds:\n  Variables:\n    x[1]=0\n  Linear/nonlinear constraints:"
-
+        @test s == "ConstraintBounds:\n  Variables:\n    NLSolversBase.UnquotedStringx[1]=0\n  Linear/nonlinear constraints:"
```
Comment (anriseth) [inline test/constraints.jl]: Is the UnquotedString stuff due to `array_eltype_show_how`? :thinking:

File: src/interface.jl (hunk 3 above)
Comment (pkofod) [reply src/interface.jl]: Yup

**Comment by pkofod** [comment]
Wohoo, nightly :+1: :+1: :+1:

**Pull request merged**